collectors/python.d.plugin/spigotmc/spigotmc.chart.py

```diff
@@ -37,6 +38,19 @@
         ]
     }
 }
+_TPS_REGEX = re.compile(
+    r'^.*: .*?'           # Message lead-in
+    r'(\d{1,2}.\d+), .*?' # 1-minute TPS value
+    r'(\d{1,2}.\d+), .*?' # 5-minute TPS value
+    r'(\d{1,2}\.\d+).*$', # 15-minute TPS value
+    re.X | re.A
+)
+_LIST_REGEX = re.compile(
+    r'^.*?'       # Message lead-in
+    r'(\d+)'      # Current user count.
+    r'.*?\d*.*$', # Rest of the line, which should include another number.
+    re.X | re.A
+)
```

Comment:
i suggest to make regex more specific

```python
In [11]: _LIST_REGEX.match('There are 0/20 players online:').groups()            
Out[11]: ('0',)

In [12]: _LIST_REGEX.match('4').groups()                                         
Out[12]: ('4',)

In [13]: _LIST_REGEX.match('always matches! 33').groups()                        
Out[13]: ('33',)
```

Reply:
if i understand it right you don't want to match any row if it has a number

Reply:
In this case, we're only pulling the first line of the output (because of the `^` and the lack of the `re.MULTILINE` switch), so we don't really need to worry beyond that line, which should look like either `There are 0/20 players online:` if the `list` command works, or `There are 0 players online on a maximum of 50.` if it doesn't and we have to use the `online` command.  If a better option can be found that can reliably match both independent of the words and formatting (because those are the most likely things to change), I'm all for it, but what's here does work to cover both cases.

Reply:
if you want to match the first number you can use `(\d+)` (search) or, if you want to be more precise, `There are (\d+)` (match).

Reply:
your regex does the same actually, it matches first group of decimals, can be simplified to `(\d+)` (search)